signac/contrib/project.py

```diff
@@ -363,7 +363,12 @@ def open_job(self, statepoint=None, id=None):
                     id = matches[0]
                 elif len(matches) > 1:
                     raise LookupError(id)
-            return self.Job(project=self, statepoint=self.get_statepoint(id), _id=id)
```

Comment:
Note that the method `get_statepoint()` is deprecated anyway, so removing it here is potentially helpful?